osu.Game.Rulesets.Osu/UI/ReplayAnalysisOverlay.cs

```diff
@@ -27,6 +27,8 @@ public partial class ReplayAnalysisOverlay : CompositeDrawable
 
         private readonly Replay replay;
 
+        private int replayFrameIndex = 0;
```

Comment:
What is this for?

Reply:
It's to keep track of the last frame that was processed for adding entries since during spectating the full replay isn't there

Reply:
The whole structure needs a reshuffle, you can't just tack on stuff like that. `initialise()` is not actually an initialisation method (it borderline wasn't before, what with it being called in `Update()`, but it *was* still mostly initialisation logic - now this method is executing *persistent* logic based on incoming frames). Yes you split the replay processing to `addEntries()` but the only caller of the method is....... `initialise()`.

Reply:
makes sense